src/UBAFeeCalculator/UBAFeeSpokeCalculator.ts

```diff
@@ -70,34 +76,44 @@ export default class UBAFeeSpokeCalculator {
-    const historicalResult = this.recentRequestFlow.slice(startIdx, endIdx).reduce((acc, flow) => {
-      let resultantValue = acc;
-      if (isUbaInflow(flow)) {
-        resultantValue = acc.add(toBN(flow.amount));
-      } else if (isUbaOutflow(flow)) {
-        resultantValue = acc.sub(toBN(flow.amount));
-      }
-
-      // If the upper trigger hurdle is surpassed, we need to return the trigger hurdle value
-      // as the running balance. This is because the trigger hurdle is the maximum value we'd like to
-      // organically grow the running balance to. If the running balance exceeds the trigger hurdle,
-      // we need to return the trigger hurdle as the running balance because at this point the dataworker
-      // will be triggered to rebalance the running balance.
-      if (upperBoundTriggerHurdle !== undefined && resultantValue.gt(upperBoundTriggerHurdle.threshold)) {
-        resultantValue = upperBoundTriggerHurdle.target;
-      }
-
-      // If the lower trigger hurdle is surpassed, we need to return the trigger hurdle value
-      // as the running balance. This is because the trigger hurdle is the minimum value we'd like to
-      // organically shrink the running balance to. If the running balance is less than the trigger hurdle,
-      // we need to return the trigger hurdle as the running balance because at this point the dataworker
-      // will be triggered to rebalance the running balance.
-      else if (lowerBoundTriggerHurdle !== undefined && resultantValue.lt(lowerBoundTriggerHurdle.threshold)) {
-        resultantValue = lowerBoundTriggerHurdle.target;
+    const historicalResult: TokenRunningBalance = this.recentRequestFlow.slice(startIdx, endIdx).reduce(
+      (acc, flow) => {
+        // If the flow is an inflow, we need to add the amount to the running balance
+        // If the flow is an outflow, we need to subtract the amount from the running balance
+        // This is reflected in the incentive balance as well
+        const resultant: TokenRunningBalance = {
+          runningBalance: acc.runningBalance[isUbaInflow(flow) ? "add" : "sub"](flow.amount),
+          incentiveBalance: acc.incentiveBalance[isUbaInflow(flow) ? "add" : "sub"](flow.amount), // TODO: Add correct incentive balance calculations
+        };
+
+        // If the upper trigger hurdle is surpassed, we need to return the trigger hurdle value
+        // as the running balance. This is because the trigger hurdle is the maximum value we'd like to
+        // organically grow the running balance to. If the running balance exceeds the trigger hurdle,
+        // we need to return the trigger hurdle as the running balance because at this point the dataworker
+        // will be triggered to rebalance the running balance.
+        if (upperBoundTriggerHurdle !== undefined && resultant.runningBalance.gt(upperBoundTriggerHurdle.threshold)) {
+          resultant.runningBalance = upperBoundTriggerHurdle.target;
```

Comment:
We'll need to store somewhere all of the net adjustments to the `resultant.runningBalance` so that the Dataworker can accumulate these in the `netSendAmount` value. Can we include these in the returned object?

Reply:
Sounds good - updated here: https://github.com/across-protocol/sdk-v2/pull/212/commits/b7fc2c429f0f67d510e40ac348b4b123f6b5cdb8